### PR TITLE
[stable/elastalert] Support referencing ES credentials from secret

### PR DESCRIPTION
**Pull request by mseiwald**
Please review @olemarkus @jertel 

#### What this PR does / why we need it:
Add support for referencing ES username/password from existing secret. This is useful for example when using Elastic Stack on Kubernetes which can automatically generate the ES credentials and store them in a secret.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/elastalert/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 description: ElastAlert is a simple framework for alerting on anomalies, spikes, or other patterns of interest from data in Elasticsearch.
 name: elastalert
-version: 1.4.0
+version: 1.4.1
```
Comment (olemarkus) [inline stable/elastalert/Chart.yaml]: You need to bump minor version here.

File: stable/elastalert/templates/deployment.yaml [hunk 2]
```diff
@@ -54,8 +54,24 @@ spec:
 {{- end }}
         resources:
 {{ toYaml .Values.resources | indent 12 }}
-{{- if .Values.optEnv }}
         env:
+{{- if .Values.elasticsearch.existingSecret }}
```
Comment (olemarkus) [inline stable/elastalert/templates/deployment.yaml]: Maybe call this `credentialsSecret` or something?

**Review by olemarkus — CHANGES_REQUESTED** [review]

File: stable/elastalert/Chart.yaml (hunk 1 above)
Comment (mseiwald) [reply stable/elastalert/Chart.yaml]: Done

File: stable/elastalert/templates/deployment.yaml (hunk 2 above)
Comment (mseiwald) [reply stable/elastalert/templates/deployment.yaml]: Done

**Pull request merged**